MetaMask/metamask-extension
# Deep clone signature tx in order to avoid state mutation bug on dev

## Explanation

This is not a functional change on the sign actions, only fixes a development issue which defined in https://github.com/MetaMask/metamask-extension/issues/17962 

Same issue has been fixed for tx signing before by @brad-decker here: https://github.com/MetaMask/metamask-extension/pull/17772/files 

## Manual Testing Steps

1. Go to test dapp
2. Click any signature
3. Open MetaMask console
4. Click Sign -- see no UI error in the console. 
If sign operation done on first click this means issue is fixed.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
